sqlmesh/core/test/definition.py

```diff
@@ -101,18 +101,30 @@ def shortDescription(self) -> t.Optional[str]:
     def setUp(self) -> None:
         """Load all input tables"""
         for table_name, values in self.body.get("inputs", {}).items():
+            all_types_are_known = False
             columns_to_types: t.Dict[str, exp.DataType] = {}
             if table_name in self.models:
                 model = self.models[table_name]
-                columns_to_types = model.columns_to_types if model.annotated else {}
+                columns_to_types = {
+                    c: t for c, t in model.columns_to_types.items() if type_is_known(t)
+                }
+                all_types_are_known = len(columns_to_types) == len(model.columns_to_types)
```

Comment:
I thought `model.columns_to_types` could be None. How come we don't need to account for this here?

Reply:
Good catch, this was an oversight. Will amend.